newrelic/hooks/external_pyzeebe.py

```diff
@@ -0,0 +1,129 @@
+                # add_attr("zeebe.client.correlationKey", correlation_key)
+            message_id = kwargs.get("message_id")
+            if message_id and len(args) > 4:
+                message_id = args[4]
+            if message_id:
+                txn._add_agent_attribute("zeebe.client.messageId", message_id)
+                # add_attr("zeebe.client.messageId", message_id)
+        elif method_name == "deploy_resource":
+            resources = list(args)
+            if len(resources) == 1 and isinstance(resources[0], (list, tuple)):
+                resources = list(resources[0])
+            if resources:
+                txn._add_agent_attribute("zeebe.client.resourceCount", len(resources))
+                # add_attr("zeebe.client.resourceCount", len(resources))
+                if len(resources) == 1:
+                    txn._add_agent_attribute("zeebe.client.resourceFile", str(resources[0]))
+                    # add_attr("zeebe.client.resourceFile", str(resources[0]))
+    except Exception:
+        _logger.warning(CLIENT_ATTRIBUTES_WARNING_LOG_MSG, exc_info=True)
+
+
+# Async wrapper that instruments router/worker annotations`
+async def _nr_wrapper_execute_one_job(wrapped, instance, args, kwargs):
+    job = args[0] if args else kwargs.get("job")
+    process_id = getattr(job, "bpmn_process_id", None) or "UnknownProcess"
+    task_type = getattr(job, "type", None) or "UnknownType"
+    txn_name = f"{process_id}/{task_type}"
+
+    with BackgroundTask(application_instance(), txn_name, group="ZeebeTask"):
+        if job is not None:
+            if hasattr(job, "key"):
+                add_custom_attribute("zeebe.job.key", job.key)
+            if hasattr(job, "type"):
+                add_custom_attribute("zeebe.job.type", job.type)
+            if hasattr(job, "bpmn_process_id"):
+                add_custom_attribute("zeebe.job.bpmnProcessId", job.bpmn_process_id)
+            if hasattr(job, "process_instance_key"):
+                add_custom_attribute("zeebe.job.processInstanceKey", job.process_instance_key)
+            if hasattr(job, "element_id"):
+                add_custom_attribute("zeebe.job.elementId", job.element_id)
+
+        result = await wrapped(*args, **kwargs)
+        return result
+
+
+# Async wrapper that instruments a ZeebeClient method.
+def _nr_client_wrapper(method_name):
+    async def wrapper(wrapped, instance, args, kwargs):
+        txn = current_transaction()
+        if txn:
+            # add_fn = add_custom_span_attribute
+            _add_client_input_attributes(method_name, txn, args, kwargs)
+            with FunctionTrace(name=method_name, group="ZeebeClient"):
+                result = await wrapped(*args, **kwargs)
+            return result
+        else:
+            # add_fn = add_custom_attribute
+            created_txn = BackgroundTask(application=application_instance(), name=method_name, group="ZeebeClient")
```

Comment:
We actually will want to avoid starting a background task here and stick to only using function tracing for these client methods. This wrapper can be edited to early exit out if no transaction is found. If one is found, then the above function tracing logic should make sense to keep.

Reply:
@umaannamalai - not sure on this - how would the client methods get picked up then if no current transaction exists? I tested only using `FunctionTrace` in the following scenario, and no data is picked up if the method where the client methods being called inside is not annotated (instrumented). I think part of the goal of the instrumentation is to allow customers to collect tracing for these methods without having to manually instrument.

Example (not picked up without using `BackgroundTask`):

```
async def main():
    ...
        
    zeebe_client = ZeebeClient(grpc_channel)
    await zeebe_client.deploy_resource('send-mail.bpmn')

asyncio.run(main())
```

Reply:
fixed in latest commit (hopefully)